lib/services/platform-service.ts

```diff
@@ -482,8 +482,34 @@ export class PlatformService implements IPlatformService {
 	}
 
 	public deployOnEmulator(platform: string, buildConfig?: IBuildConfig): IFuture<void> {
-		this.$options.emulator = true;
-		return this.deployOnDevice(platform, buildConfig);
+		platform = platform.toLowerCase();
+
+		if (this.$options.availableDevices) {
+			return (() => {
+				let exec = require("child_process").exec;
+
+				if (this.$mobileHelper.isiOSPlatform(platform)) {
```

Comment:
Instead of branching here you can use something like:
```
let platformData = this.$platformsData.getPlatformData(platform); 
platformData.platformProjectService.listDevices().wait();
```
And add the listDevices to the project interface and implement as necessary for the iOS and Android project services.